lib/adapters/command-execution-adapter.ts

```diff
@@ -0,0 +1,30 @@
+import { ExecuteCommandParams, ServerCapabilities } from "../languageclient";
+import { LanguageClientConnection } from "../main";
+
+export type CommandCustomCallbackFunction = (command: ExecuteCommandParams) => Promise<any | void>;
+
+export default class CommandExecutionAdapter {
+    private static commandsCustomCallbacks: Map<string, CommandCustomCallbackFunction> = new Map<string, CommandCustomCallbackFunction>();
+
+    public static canAdapt(serverCapabilities: ServerCapabilities): boolean {
+      return serverCapabilities.executeCommandProvider != null;
```

Comment:
if `serverCapabilities.executeCommandProvider` is `false` this would return `true`. Shouldn't we just check for `true`?
```suggestion
      return serverCapabilities.executeCommandProvider === true;
```

Reply:
`executeCommandProvider` is of type `ExecuteCommandOptions`unlike other providers that are either boolean or an object of options (depending on protocol version). I think the check is the only way of knowing if it was supplied or not.

Reply:
I see